build/toolchain/gcc_toolchain.gni

```diff
@@ -149,7 +150,6 @@ template("gcc_toolchain") {
     toolchain_args = {
       # Populate toolchain args from the invoker.
       forward_variables_from(invoker_toolchain_args, "*")
-      build_with_separate_cobalt_toolchain = build_with_separate_cobalt_toolchain
```

Comment:
You instead should be passing through is_modular_toolchain to propagate it to other toolchains

Reply:
the variable is only global to the toolchain, it doesn't propagate to the other toolchains unless passed to them

Reply:
i dont understand, GN normally complains if the variable is not needed/ not set . None of he tests fail in the GN step because of this.

Reply:
> GN normally complains if the variable is not needed/ not set

The variable is needed and is set, it's just a question of what it's set to. It works now as (I believe) it is implicitly passing the variable from the outer scope to the scope captured by `toolchain_args`—so it is set correctly, but brittle and it's better to be explicit about passing the same value through

See https://gn.googlesource.com/gn/+/master/docs/reference.md#toolchain-overview for more info